controllers/reclaimspacecronjob_controller.go

```diff
@@ -0,0 +1,370 @@
+	err := r.Client.Get(ctx, req.NamespacedName, rsCronJob)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			logger.Info("ReclaimSpaceCronJob resource not found")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// set history limit defaults, if not specified.
+	if rsCronJob.Spec.FailedJobsHistoryLimit == nil {
+		*rsCronJob.Spec.FailedJobsHistoryLimit = defaultFailedJobsHistoryLimit
+	}
+	if rsCronJob.Spec.SuccessfulJobsHistoryLimit == nil {
+		*rsCronJob.Spec.SuccessfulJobsHistoryLimit = defaultSuccessfulJobsHistoryLimit
+	}
+
+	var childJobs csiaddonsv1alpha1.ReclaimSpaceJobList
+	if err := r.List(ctx, &childJobs, client.InNamespace(req.Namespace), client.MatchingFields{jobOwnerKey: req.Name}); err != nil {
+		logger.Error(err, "Failed to list child ReclaimSpaceJobs")
+		return ctrl.Result{}, err
+	}
+
+	// find the active,failed and successful list of jobs
+	var activeJob *csiaddonsv1alpha1.ReclaimSpaceJob
+	var successfulJobs, failedJobs []*csiaddonsv1alpha1.ReclaimSpaceJob
+	var mostRecentTime, lastSuccessfulTime *time.Time // find the last run so we can update the status
+
+	for i, job := range childJobs.Items {
+		switch job.Status.Result {
+		case "": // ongoing
+			activeJob = &childJobs.Items[i]
+		case csiaddonsv1alpha1.OperationResultFailed:
+			failedJobs = append(failedJobs, &childJobs.Items[i])
+		case csiaddonsv1alpha1.OperationResultSucceeded:
+			successfulJobs = append(successfulJobs, &childJobs.Items[i])
+			completionTime := childJobs.Items[i].Status.CompletionTime
+			if lastSuccessfulTime == nil {
+				lastSuccessfulTime = &completionTime.Time
+			} else if completionTime != nil && lastSuccessfulTime.Before(completionTime.Time) {
+				lastSuccessfulTime = &completionTime.Time
+			}
+		}
+
+		// reconstitute scheduled time from annotation.
+		scheduledTimeForJob, err := getScheduledTimeForRSJob(&job)
+		if err != nil {
+			logger.Error(err, "Failed to parse schedule time for child ReclaimSpaceJob", "ReclaimSpaceJob", job.Name)
+			continue
+		}
+		if scheduledTimeForJob != nil {
+			if mostRecentTime == nil {
+				mostRecentTime = scheduledTimeForJob
+			} else if mostRecentTime.Before(*scheduledTimeForJob) {
+				mostRecentTime = scheduledTimeForJob
+			}
+		}
+	}
+
+	if mostRecentTime != nil {
+		rsCronJob.Status.LastScheduleTime = &metav1.Time{Time: *mostRecentTime}
+	} else {
+		rsCronJob.Status.LastScheduleTime = nil
+	}
+	if lastSuccessfulTime != nil {
+		rsCronJob.Status.LastSuccessfulTime = &metav1.Time{Time: *lastSuccessfulTime}
+	}
+	rsCronJob.Status.Active = nil
+	if activeJob != nil {
+		jobRef, err := ref.GetReference(r.Scheme, activeJob)
+		if err != nil {
+			logger.Error(err, "Failed to make reference to active job", "job", activeJob)
+		}
+		rsCronJob.Status.Active = jobRef
+	}
+	logger.Info("ReclaimSpaceJob count", "successfulJobs", len(successfulJobs), "failedJobs", len(failedJobs))
```

Comment:
job successful and failed job names?

Reply:
> job successful and failed job names?

The number of job maybe large, 
the child jobs have name = `<cronjob-name>-<time-hash>`
so it is easy to identify child jobs
```
[rakshith@fedora kubernetes-csi-addons]$ k get reclaimspacejobs.csiaddons.openshift.io 
NAME                                    NAMESPACE   AGE    RETRIES   RESULT
reclaimspacecronjob-sample-1642503300   rook-ceph   109m             Succeeded
reclaimspacecronjob-sample-1642503360   rook-ceph   108m             Succeeded
reclaimspacecronjob-sample-1642503420   rook-ceph   107m             Succeeded
```
I don't think printing names is required

Reply:
in that case, IMO logging just numbers is not useful at all. as it won't provide information like which all are failed and succeed.

Reply:
removing it